sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -49,7 +49,7 @@ import org.apache.spark.util.Utils
 object ExpressionEncoder {
 
   def apply[T : TypeTag](): ExpressionEncoder[T] = {
-    apply(ScalaReflection.encoderFor[T])
+    apply(ScalaReflection.encoderForWithRowEncoderSupport[T])
```

Comment:
What happens when we use the old method?

Reply:
Nothing will break. I changed this one because it's from internal code, so just to be safe.

Reply:
But it does not work right? I would try it :)... The problem is that you cannot create an ExpressionEncoder for an UnboundRowEncoder. You need a schema to build an expression encoder.